caffe-grid/pom.xml

```diff
@@ -18,11 +18,28 @@
     <maven.compiler.target>1.7</maven.compiler.target>
     <hadoop.version>2.5.0</hadoop.version>
     <scala.version>2.10.4</scala.version>
```

Comment:
This turns out to be a difficult problem.  The general approach is:

          <plugin>
            <groupId>org.codehaus.mojo</groupId>
            <artifactId>build-helper-maven-plugin</artifactId>
            <version>1.10</version>
            <executions>
                <execution>
                    <id>regex-property</id>
                    <goals>
                        <goal>regex-property</goal>
                    </goals>
                    <configuration>
             <regexPropertySettings>
                <regexPropertySetting>
                        <name>scala.major.version</name>
                        <value>${scala.version}</value>
                        <regex>\.[\d]+$</regex>
                        <replacement></replacement>
                        <failIfNoMatch>true</failIfNoMatch>
                </regexPropertySetting>
             </regexPropertySettings>
                    </configuration>
                </execution>
            </executions>
	</plugin>

But it is not working.    I do not have more bandwidth presently on  this.  Using the build-helper-maven-plugin is new to me and I do not have visibility into why the above does not work.

I feel this is not a showstopper.  Let's leave this as a TODO: it is not worth to halt the entire PR.